src/main/java/net/snowflake/ingest/streaming/internal/DataValidationUtil.java

```diff
@@ -1055,6 +1055,82 @@ static int validateAndParseBoolean(String columnName, Object input, long insertR
         insertRowIndex);
   }
 
+  /**
+   * Validate and cast Iceberg struct column to Map<String, Object>. Allowed Java type:
+   *
+   * <ul>
+   *   <li>Map<String, Object>
+   * </ul>
+   *
+   * @param columnName Column name, used in validation error messages
+   * @param input Object to validate and parse
+   * @param insertRowIndex Row index for error reporting
+   * @return Object cast to Map
+   */
+  static Map<String, ?> validateAndParseIcebergStruct(
+      String columnName, Object input, long insertRowIndex) {
+    if (!(input instanceof Map)) {
+      throw typeNotAllowedException(
+          columnName,
```

Comment:
will columnName here be the full path (for a nested structured object with a failure deep in the hierarchy)

Reply:
Passed full path into `DataValidationUtil` in Iceberg mode.